source/common/json/json_loader.cc

```diff
@@ -743,6 +743,16 @@ ObjectSharedPtr Factory::loadFromYamlString(const std::string& yaml) {
     return parseYamlNode(YAML::Load(yaml));
   } catch (YAML::ParserException& e) {
     throw EnvoyException(e.what());
+  } catch (YAML::BadConversion& e) {
+    throw EnvoyException(e.what());
+  } catch (...) {
+    // There is a potentially wide space of exceptions thrown by the YAML parser,
+    // and enumerating them all may be difficult. Envoy doesn't work well with
+    // unhandled exceptions, so we capture them and record the exception name in
+    // the Envoy Exception text.
+    std::exception_ptr p = std::current_exception();
+    throw EnvoyException(std::string("Unexpected YAML exception: ") +
+                         p.__cxa_exception_type()->name());
```

Comment:
This `__cxa_exception_type` thing seems kind of scary; can you discuss how it works and any portability (compiler, libstdc++/libc++) concerns?

Reply:
fixed.